src/integrationtests/java/com/aws/greengrass/integrationtests/certificateauthority/OfflineAuthenticationTest.java

```diff
@@ -0,0 +1,155 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.lenient;
+
+@ExtendWith({MockitoExtension.class, GGExtension.class})
+public class OfflineAuthenticationTest {
+    @Mock
+    SecurityService securityServiceMock;
+    @Mock
+    private GreengrassServiceClientFactory clientFactory;
+    @Mock
+    private GroupManager groupManager;
+    @Mock
+    CertificateExpiryMonitor certExpiryMonitorMock;
+    @Mock
+    CISShadowMonitor cisShadowMonitorMock;
+    @Mock
+    private GreengrassV2DataClient client;
+    @TempDir
+    Path rootDir;
+    private Kernel kernel;
+    private IotAuthClientFake iotAuthClientFake;
+
+
+    @BeforeEach
+    void setup(ExtensionContext context) throws DeviceConfigurationException {
+        ignoreExceptionOfType(context, SpoolerStoreException.class);
+
+        // Set this property for kernel to scan its own classpath to find plugins
+        System.setProperty("aws.greengrass.scanSelfClasspath", "true");
+        kernel = new Kernel();
+        kernel.getContext().put(GroupManager.class, groupManager);
+        kernel.getContext().put(SecurityService.class, securityServiceMock);
+        kernel.getContext().put(CertificateExpiryMonitor.class, certExpiryMonitorMock);
+        kernel.getContext().put(CISShadowMonitor.class, cisShadowMonitorMock);
+        kernel.getContext().put(GreengrassServiceClientFactory.class, clientFactory);
+
+        DomainEvents domainEvents =  new DomainEvents();
+        CertificateStore certificateStore = new CertificateStore(rootDir, domainEvents , securityServiceMock);
+
+        kernel.getContext().put(DomainEvents.class, domainEvents);
+        kernel.getContext().put(CertificateStore.class, certificateStore);
+
+        iotAuthClientFake = new IotAuthClientFake();
+        kernel.getContext().put(IotAuthClient.class, iotAuthClientFake);
+
+        lenient().when(clientFactory.fetchGreengrassV2DataClient()).thenReturn(client);
+    }
+
+    @AfterEach
+    void cleanup() {
+        kernel.shutdown();
+    }
+
+    private void givenNucleusRunningWithConfig(String configFileName) throws InterruptedException {
+        CountDownLatch authServiceRunning = new CountDownLatch(1);
+        kernel.parseArgs("-r", rootDir.toAbsolutePath().toString(), "-i",
+                getClass().getResource(configFileName).toString());
+        kernel.getContext().addGlobalStateChangeListener((service, was, newState) -> {
+            if (ClientDevicesAuthService.CLIENT_DEVICES_AUTH_SERVICE_NAME.equals(service.getName()) && service.getState()
+                    .equals(State.RUNNING)) {
+                authServiceRunning.countDown();
+            }
+        });
+        kernel.launch();
+        assertThat(authServiceRunning.await(30L, TimeUnit.SECONDS), is(true));
+    }
+
+    @Test
+    void GIVEN_clientDevice_WHEN_verifyingItsIdentity_THEN_pemStored() throws InterruptedException,
```

Comment:
This is le test that shows what we are currently doing. Note that I have not handled removing PEMs from the store